kotlin/display-utility-associations/src/main/java/com/esri/arcgisruntime/sample/displayutilityassociations/MainActivity.kt

```diff
@@ -0,0 +1,173 @@
+    utilityNetwork.addDoneLoadingListener {
+
+      // get all of the edges and junctions in the network
+      val edges =
+        utilityNetwork.definition.networkSources.filter { it.sourceType == UtilityNetworkSource.Type.EDGE }
+      val junctions =
+        utilityNetwork.definition.networkSources.filter { it.sourceType == UtilityNetworkSource.Type.JUNCTION }
+
+      // add all edges that are not subnet lines to the map
+      edges.filter { it.sourceUsageType != UtilityNetworkSource.UsageType.SUBNET_LINE }
+        .forEach { source ->
+          mapView.map.operationalLayers.add(FeatureLayer(source.featureTable))
+        }
+      // add all junctions to the map
+      junctions.forEach { source ->
+        mapView.map.operationalLayers.add(FeatureLayer(source?.featureTable))
+      }
+
+      // populate the legend in the UI
+      val attachmentSwatchFuture = attachmentSymbol.createSwatchAsync(this, Color.TRANSPARENT)
+      attachmentSwatch.setImageBitmap(attachmentSwatchFuture.get())
+      val connectSwatchFuture = connectivitySymbol.createSwatchAsync(this, Color.TRANSPARENT)
+      connectivitySwatch.setImageBitmap(connectSwatchFuture.get())
+
+      // add association graphics at the initial view point
+      addAssociationGraphicsAsync()
+
+      // listen for navigation changes
+      mapView.addNavigationChangedListener {
+        // add association graphics for viewpoint after navigation change
+        addAssociationGraphicsAsync()
+      }
+    }
+  }
+
+  /**
+   * If greater than the maxScale, add association graphics for the map view's current extent.
+   *
+   */
+  private fun addAssociationGraphicsAsync() {
+
+    // check if the current viewpoint is outside of the max scale
+    if (mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE).targetScale >= maxScale) {
+      return
+    }
+
+    // check if the current viewpoint has an extent
+    (mapView.getCurrentViewpoint(Viewpoint.Type.BOUNDING_GEOMETRY).targetGeometry.extent)?.let { extent ->
+      // get all of the associations in extent
+      val associationsFuture = utilityNetwork.getAssociationsAsync(extent)
+      associationsFuture.addDoneListener {
+        val associations = associationsFuture.get()
+
+        associations.forEach { association ->
+          // if the graphics overlay doesn't already contain the association
+          if (!associationsOverlay.graphics.any {
+              it.attributes.containsKey("GlobalId") && UUID.fromString(
+                it.attributes["GlobalId"].toString()
+              ) == association.globalId
+            }) {
+
+            // add a graphic for the association
+            var symbol: SimpleLineSymbol? = null
+            if (association.associationType == UtilityAssociationType.ATTACHMENT) {
+              symbol = attachmentSymbol
+            } else if (association.associationType == UtilityAssociationType.CONNECTIVITY) {
+              symbol = connectivitySymbol
+            }
```

Comment:
Maybe this could read easier with a when statement? ie.

```suggestion
            var symbol: SimpleLineSymbol? = when (association.associationType) {
              UtilityAssociationType.ATTACHMENT -> attachmentSymbol
              UtilityAssociationType.CONNECTIVITY -> connectivitySymbol
              else -> null
            }
```

Reply:
Oh yeah, much better. Thanks!